### PR TITLE
docs: [IAC-2518]: IaCM Best Practices

### PR DESCRIPTION
**Pull request by richardblack-Harness**
## Description

* New reference doc to highlight the best practices for IaCM usage
* [JIRA](https://harness.atlassian.net/browse/IAC-2518)

### Screenshot / Preview
https://github.com/user-attachments/assets/a48a4028-bd9c-4844-9773-048d805dc2c5

PRs must meet these requirements to be merged:
- [x] Successful preview build.
- [x] Code owner review.
- [x] No merge conflicts.
- [x] Release notes/new features docs: Feature/version released to at least one prod environment.

### CODE REVIEW

File: kb/reference-architectures/iacm/iacm-best-practices.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+import HarnessApiData from '../../../src/components/HarnessApiData/index.tsx';
+
+Harness Infrastructure as Code allows you to define, deploy, and manage infrastructure across environments, ensuring compliance and control. Key features include cost estimation, approval steps, PR automation, policy enforcement, and drift detection, which can integrate seamlessly with other Harness modules and third-party services, enhancing your DevOps lifecycle.
+
+This document provides a set of best practices and guidelines aimed at helping teams implement and manage IaCM effectively. It serves as a reference to navigate the complexities of infrastructure management, offering clear, actionable recommendations to optimize performance, enhance security, and prevent common pitfalls.
+
+## Workflow hierarchy
+
+- **Account** is the root level, containing multiple Projects.
+    - **Projects** contain multiple Workspaces.
```
Comment (rssnyder) [inline kb/reference-architectures/iacm/iacm-best-practices.md:15]: probably need to include orgs here
```suggestion
- **Account** is the root level, containing multiple Organizations.
    - **Organizations** contain multiple Projects.
        - **Projects** contain multiple Workspaces.
```

File: kb/reference-architectures/iacm/iacm-best-practices.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        - **Workspaces** are isolated environments for IaC execution within a Project.
+- **Pipelines** are independent entities that can be executed against any Workspace within a Project.
```
Comment (rssnyder) [inline kb/reference-architectures/iacm/iacm-best-practices.md:17]: Should this go under projects, next to Workspaces?

File: kb/reference-architectures/iacm/iacm-best-practices.md (hunk 1 above)
Comment (richardblack-Harness) [reply kb/reference-architectures/iacm/iacm-best-practices.md:15]: Missed orgs, good spot.

File: kb/reference-architectures/iacm/iacm-best-practices.md (hunk 2 above)
Comment (richardblack-Harness) [reply kb/reference-architectures/iacm/iacm-best-practices.md:17]: Maybe, I was in two minds about Pipeline placement, as they are a separate entity but ran against workspaces. Alongside workspace probably would work. I've asked Kseniia to create a diagram for this section that should help clarify this as well.

File: kb/reference-architectures/iacm/iacm-best-practices.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+<!-- placeholder for hierarchy diagram -->
+
+### IaCM approaches / options
+- Terraform / OpenTofu
+- Harness UI
+- Harness API (with link to API docs and also Postman collection)
```
[the remark this answers is not in the document]
Comment (richardblack-Harness) [reply kb/reference-architectures/iacm/iacm-best-practices.md:24]: Perhaps a more usable example would work here is a tabbed view. Like creating a workspace with each approach. I'm not sure how much each approach is used by clients. While we use the API, do users ever use it?? I should add a downloadable Postman collection in here too.

[the remark this answers is not in the document]
Comment (richardblack-Harness) [reply kb/reference-architectures/iacm/iacm-best-practices.md:24]: Removing for now

File: kb/reference-architectures/iacm/iacm-best-practices.md [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+    fallback=""
+    parse='.[-1] | " (latest: v\(.))"'></HarnessApiData>.
+    
+ For **Terraform**, we support all MPL versions up to **1.5.x**, any BSL versions (from 1.6.0) are not supported.
+:::
+
+### Configuration Guidelines
+
+As a first step, we recommend configuring your Cloud Provider and Code repository **connectors** to help streamline further configurations like workspace and pipeline creation so they can be easily selected.  
+
```
Comment (rssnyder) [inline kb/reference-architectures/iacm/iacm-best-practices.md:36]:
```suggestion
```

**Review by rssnyder — APPROVED** [review]
thanks for addressing the feedback!

File: kb/reference-architectures/iacm/iacm-best-practices.md [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+
```
Comment (RickCraig) [inline kb/reference-architectures/iacm/iacm-best-practices.md:37]: Double space, no biggy

File: kb/reference-architectures/iacm/iacm-best-practices.md [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+## Security
+- **Access Controls:** [Role-based access control (RBAC)](https://developer.harness.io/docs/platform/role-based-access-control/rbac-in-harness/) lets you control who can access your resources and what actions they can perform on the resources. To do this, a Harness account administrator assigns resource-related permissions to members of user groups.
+- **Secret Management:** Go to the [secret management page](https://developer.harness.io/docs/category/secrets-management) to see all supported secret management option available in the Harness Platform and determine what option is best suited for your needs. As mentioned above, Harness offer integration with multiple secret management options but recommend [Harness secret manager](https://developer.harness.io/docs/platform/secrets/secrets-management/harness-secret-manager-overview/) to help offer optimal performance.
+- **OPA Policies:** Use [OPA policies](https://developer.harness.io/docs/platform/governance/policy-as-code/harness-governance-overview/) to implement governance and trigger pipeline warnings or failures when policy conditions are not met. 
+-
```
Comment (RickCraig) [inline kb/reference-architectures/iacm/iacm-best-practices.md:66]: What's this wee dash here?

**Review by RickCraig — APPROVED** [review]

File: kb/reference-architectures/iacm/iacm-best-practices.md (hunk 6 above)
Comment (richardblack-Harness) [reply kb/reference-architectures/iacm/iacm-best-practices.md]: Removed

**Review by rssnyder — APPROVED** [review]
+formatting

**Review by RajiKoppala — APPROVED** [review]

File: src/components/ReferenceArchitectures/data/referenceArchitecture.ts [hunk 7]
```diff
@@ -9,6 +9,13 @@ export const ReferenceArchitecturesCards: CardSections = [
     name: "Get started with Reference Architectures",
     description: "",
     list: [
+      {
+        title: "Infrastructure as Code Management: Best Practices",
+        module: MODULES.platform,
```
Comment (wei-harness) [inline src/components/ReferenceArchitectures/data/referenceArchitecture.ts:14]: Ideally, the module should be `MODULES.iacm`, but it doesn't break anything for now.

**Review by wei-harness — APPROVED** [review]
LGTM.
also added an inline comment.

File: src/components/ReferenceArchitectures/data/referenceArchitecture.ts (hunk 7 above)
Comment (richardblack-Harness) [reply src/components/ReferenceArchitectures/data/referenceArchitecture.ts:14]: Fair point, I'll update this in another PR that doesn't require so many approvals. Thanks Wei.

**Pull request merged**